resolver/src/lookup.rs

```diff
@@ -445,6 +471,21 @@ pub mod tests {
         );
     }
 
+    #[test]
+    fn test_lookup_ttl() {
+        assert_eq!(
+            InnerLookupFuture::lookup(
+                vec![Name::root()],
+                RecordType::A,
+                DnsRequestOptions::default(),
+                CachingClient::new(0, mock(vec![v4_message()])),
+            ).wait()
+                .unwrap()
+                .ttl(),
+            Some(Duration::from_secs(86400))
```

Comment:
It would be good to have a test for CNAME, in particular where the CNAME expires earlier than the A records.

It would also be good to have a test for NXDOMAIN that shows how this works for NXDOMAIN. In particular, "The remaining of the current meanings, of being the TTL to be used for negative responses, is the new defined meaning of the SOA minimum field." from https://tools.ietf.org/html/rfc2308; see https://tools.ietf.org/html/rfc2308#section-5.